homeassistant/components/light/mqtt.py

```diff
@@ -457,6 +540,30 @@ def async_turn_on(self, **kwargs):
                 self._hs = kwargs[ATTR_HS_COLOR]
                 should_update = True
 
+        if ATTR_HS_COLOR in kwargs and \
+           self._topic[CONF_HUE_COMMAND_TOPIC] is not None and \
+           self._topic[CONF_SATURATION_COMMAND_TOPIC] is not None:
+
+            hs_color = kwargs[ATTR_HS_COLOR]
+            tpl = self._templates[CONF_HUE_COMMAND_TEMPLATE]
+            if tpl:
+                hue_value = int(tpl.async_render_with_possible_json_value(
+                    str(hs_color[0])))
```

Comment:
this is completely wrong. you can't convert the passed in color to a string and say it's possible JSON value?